dts/arm/silabs/efr32bg22c224f512im40.dtsi

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <silabs/efr32bg22.dtsi>
+
+/ {
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(32)>;
+	};
+
+	soc {
+		compatible = "silabs,efr32bg22c224f512im40", "silabs,efr32bg22",
+						"silabs,efr32", "simple-bus";
+
+		flash-controller@40030000 {
+			flash0: flash@0 {
+				reg = <0 DT_SIZE_K(512)>;
+			};
+		};
+	};
+};
```

Comment:
why not extending using nodelabels, `&flash0`, `sram0`? see GD32 for example

Reply:
Fixed